--- a/net/minecraft/network/protocol/game/ServerGamePacketListener.java
+++ b/net/minecraft/network/protocol/game/ServerGamePacketListener.java
@@ -22,6 +_,7 @@
    void m_7951_(ServerboundContainerClosePacket p_133750_);
 
    void m_7423_(ServerboundCustomPayloadPacket p_133751_);
+   void handleCustomPayloadLunamura(ServerboundCustomPayloadPacket p_133751_);
 
    void m_6946_(ServerboundInteractPacket p_133754_);
 
